1.2/server/main.py

```diff
@@ -0,0 +1,21 @@
+from flask import Flask, flash, redirect, render_template, \
+     request, url_for
+
+app = Flask(__name__)
+app.secret_key = b'_5#y2L"F4Q8z\n\xec]/'
+
+@app.route('/')
+def index():
+    return render_template('index.html')
+
+@app.route('/login', methods=['GET', 'POST'])
+def login():
+    error = None
+    if request.method == 'POST':
+        if request.form['username'] != 'admin' or \
+                request.form['password'] != 'secret':
+            error = 'Invalid credentials'
+        else:
+            flash('You were successfully logged in')
+            return redirect(url_for('index'))
+    return render_template('login.html', error=error)
```

Comment:
This line is missing at the last line
```
app.run()
```

Reply:
Got it. Corrected it and raised a PR. Please check.